src/sentry/static/sentry/app/views/projectPluginDetails.jsx

```diff
@@ -0,0 +1,181 @@
+import React from 'react';
+import AsyncView from './asyncView';
+import {t} from '../locale';
+import PluginConfig from '../components/pluginConfig';
+import ExternalLink from '../components/externalLink';
+import IndicatorStore from '../stores/indicatorStore';
+
+export default class ProjectPlugins extends AsyncView {
+  getTitle() {
+    let {plugin} = this.state;
+    if (plugin && plugin.name) {
+      return plugin.name;
+    } else {
+      return 'Sentry';
+    }
+  }
+
+  getEndpoints() {
+    let {projectId, orgId, pluginId} = this.props.params;
+    return [['plugin', `/projects/${orgId}/${projectId}/plugins/${pluginId}/`]];
+  }
+
+  trimSchema(value) {
+    return value.split('//')[1];
+  }
+
+  resetConfiguration() {
+    let {projectId, orgId, pluginId} = this.props.params;
+    let loadingIndicator = IndicatorStore.add(t('Saving changes..'));
+    this.api.request(`/projects/${orgId}/${projectId}/plugins/${pluginId}/reset/`, {
+      method: 'POST',
+      success: plugin => {
+        this.setState({plugin});
+      },
+
+      complete: () => IndicatorStore.remove(loadingIndicator),
+    });
+  }
+
+  enable() {
+    this.toggleEnable(true);
+  }
+
+  disable() {
+    this.toggleEnable(false);
+  }
+
+  handleDisable() {
+    this.setState({plugin: Object.assign({}, this.state.plugin, {enabled: false})});
+  }
+
+  toggleEnable(shouldEnable) {
+    let method = shouldEnable ? 'POST' : 'DELETE';
+
+    let {orgId, projectId, pluginId} = this.props.params;
+
+    let loadingIndicator = IndicatorStore.add(t('Saving changes..'));
+
+    this.api.request(`/projects/${orgId}/${projectId}/plugins/${pluginId}/`, {
+      method,
+      success: () => {
+        this.setState({
+          plugin: Object.assign({}, this.state.plugin, {enabled: shouldEnable}),
+        });
+      },
+      complete: () => IndicatorStore.remove(loadingIndicator),
+    });
+  }
+
+  renderActions() {
+    let {plugin} = this.state;
+
+    let reset = (
+      <button
+        type="submit"
+        className="btn btn-default"
+        onClick={() => this.resetConfiguration()}
+      >
+        {t('Reset Configuration')}
+      </button>
+    );
+
+    let enable = (
+      <button
+        type="submit"
+        className="btn btn-default"
+        onClick={() => this.enable()}
+        style={{marginRight: '6px'}}
+      >
+        {t('Enable Plugin')}
+      </button>
+    );
+
+    let disable = (
+      <button
+        type="submit"
+        className="btn btn-danger"
+        onClick={() => this.disable()}
+        style={{marginRight: '6px'}}
+      >
+        {t('Disable Plugin')}
+      </button>
+    );
+
+    let toggleEnable = plugin.enabled ? disable : enable;
+
+    return (
+      <div className="pull-right">
+        {plugin.canDisable && toggleEnable}
+        {reset}
```

Comment:
nit/opinion: I think it's easier to read when we group JSX together. If I'm looking at the return value I don't know what `reset` is (similarly with `toggleEnable`, but with that there's some logic involved). I think either removing the unneeded `reset` var, or giving it a more descriptive var name would help, or create a convention where these var names should be in Capital letters.